.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,79 @@
+name: Unit and linting tests
+
+on:
+  workflow_call:
+    inputs:
+      plugin:
+        type: string
+jobs:
+  unit-tests:
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python ${{matrix.python-version}}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{matrix.python-version}}
+          cache: 'pip'
```

Comment:
<img width="746" alt="image" src="https://user-images.githubusercontent.com/14007150/230417091-42756bbb-dcd1-4615-8a9d-aba78c03248a.png">

All of the plugins are hitting the same cache key I think, which is why it's not really helping.

Reply:
Added cache step with different keys for each plugin